R/user-record.R

```diff
@@ -0,0 +1,30 @@
+#' Return Crunch User Information
+#'
+#' This function returns information about the current user of the shiny app. This
+#' is useful if you want to change the behavior of the app depending on who is
+#' viewing the app. You can access elements of this record with crunch functions
+#' like `name()` or `email().`
+#' 
+#' @param text_error If TRUE the function will return a character vector instead of
+#' erroring. This is useful for handling the error in your shiny app.  
+#' @return A user record if the user is logged in, otherwise a character vector or error
+#' @export
+shinyUser <- function (text_error = TRUE) .buildReactiveExpr(".getUserRecord", text_error)
```

Comment:
I don't think this text_error is quite right. The example apps are a little quirky: one checks for whether the object is character, the other checks for whether it is a UserEntity. It works, but I don't think it's the interface we want to get people to use. 

I know we have https://github.com/Crunch-io/crunchy/issues/10 open but let me see if I can come up with a cleaner interim solution.